nemo/collections/llm/modelopt/quantization/quantizer.py

```diff
@@ -288,7 +293,8 @@ def quantize(self, model: "MegatronParallel", forward_loop=None):
 
         If forward_loop is not provided, a forward loop will be created using the calibration dataset.
         """
-        if forward_loop is None:
+        modelopt_cfg = QUANT_CFG_CHOICES[self.quantization_config.algorithm]
+        if forward_loop is None and mtq.config.need_calibration(modelopt_cfg):
```

Comment:
That's by far the best solution. Now details remain:
1. Using `algorithm = None` a.k.a. `"no_quant"`
2. Loading config from a disk.

(Both cases are tested in the internal CI system, but not here on Github.)

To address these corner cases it's enough to move the check after L310 as:
```
        if forward_loop is None and mtq.config.need_calibration(quant_cfg):
            forward_loop = self._get_forward_loop(model)
```
see also git diff attached with the propsal:

![Zrzut ekranu 2025-06-13 103701](https://github.com/user-attachments/assets/86b2cc14-e258-4102-898e-18872acaa06d)

Reply:
But since the Github CI is currently halfway through, let's change that in a subsequent MR, no need to edit this one anymore. And again thanks for coming with the `mtq.config.need_calibration` method here.